packages/client/auth-core/src/services/CrossmintService.ts

```diff
@@ -0,0 +1,59 @@
+import { CROSSMINT_PROD_URL, CROSSMINT_STG_URL, CrossmintEnvironment } from "../utils";
+
+export type FetchCrossmintParams = {
+    endpoint: string;
+    options: {
+        method: "GET" | "POST" | "PUT" | "DELETE";
+        body?: any;
+    };
+    onServerErrorMessage?: string;
+};
+
+export class CrossmintService {
+    protected crossmintAPIHeaders: Record<string, string>;
+    public readonly crossmintBaseUrl: string;
+    private static urlMap: Record<CrossmintEnvironment, string> = {
+        staging: CROSSMINT_STG_URL,
+        production: CROSSMINT_PROD_URL,
+    };
+
+    constructor(apiKey: string, jwtToken: string | null, environment?: CrossmintEnvironment) {
+        this.crossmintAPIHeaders = {
+            accept: "application/json",
+            "content-type": "application/json",
+            authorization: `Bearer ${jwtToken}`,
+            "x-api-key": apiKey,
+        };
+        this.crossmintBaseUrl = this.getUrlFromEnv(environment ?? "production");
+    }
+
+    async fetchCrossmintAPI({ endpoint, options = { method: "GET" }, onServerErrorMessage }: FetchCrossmintParams) {
```

Comment:
Can we use Swagger / OpenAPI to generate a better client for our APIs?

Reply:
The `fetchCrossmintAPI` was thought for internal purposes. As of now, this will be used by external devs, this function doesn't make sense anymore.